jpa/spi/src/main/java/org/jipijapa/plugin/spi/PersistenceUnitService.java

```diff
@@ -38,6 +38,12 @@
      */
     EntityManagerFactory getEntityManagerFactory();
 
+    /**
+     * Returns the configured EntityManager cache.
+     * @return
+     */
+    EntityManagerCache getEntityManagerCache();
```

Comment:
I'm not sure that an implementing class would ever just return null, as it more likely that an implementation would conditionally return null if the cache is empty, otherwise an EntityManager instance.

Reply:
Is it expected that say EclipseLink would implement this?

Reply:
No, EclipseLink would not ever implement this spi.  Your questions got me thinking, that this really isn't an SPI but should instead be a non-spi interface.